presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarImplementation.java

```diff
@@ -412,8 +417,8 @@ private void parseArguments(Method method)
                     checkArgument(annotations.length == 1, "Meta parameters may only have a single annotation [%s]", method);
                     checkArgument(argumentTypes.isEmpty(), "Meta parameter must come before parameters [%s]", method);
                     Annotation annotation = annotations[0];
-                    if (annotation instanceof TypeParameter) {
-                        checkArgument(typeParameters.contains(annotation), "Injected type parameters must be declared with @TypeParameter annotation on the method [%s]", method);
+                    if (annotation instanceof TypeParameter && !typeParameters.contains(annotation)) {
```

Comment:
Is ` && !typeParameters.contains(annotation)` necessary? It looks like `checkTypeParameters` will naturally do the same thing.